vcs/src/main/java/org/openjdk/skara/vcs/git/GitRepository.java

```diff
@@ -1720,4 +1720,13 @@ private Diff diffStaged() throws IOException {
             throw t;
         }
     }
+
+    @Override
+    public boolean isFileUpdated(String filename, Hash h) throws IOException {
+        return commits().asList().stream()
+                .filter(commit -> commit.hash().equals(h))
+                .anyMatch(commit -> commit.parentDiffs().stream()
+                        .anyMatch(diff -> diff.patches().stream()
+                                .anyMatch(patch -> patch.toString().contains(filename))));
+    }
```

Comment:
@erikj79 I am not sure I get you correctly. Correct me if I misunderstand what you mean.

Reply:
I wasn't very clear. I don't think we need this method on GitRepository. I figured we could just inline this at the single call site in CheckRun, but since the expression is quite long, then probably a private method in CheckRun would be best.

You can use `commits(String range)` to just get the single commit we are interested in.